cmd/cluster-agent/api/agent/agent.go

```diff
@@ -148,11 +148,12 @@ func getConfigCheck(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	configs := common.AC.GetLoadedConfigs()
 	configSlice := make([]integration.Config, 0)
-	for _, config := range configs {
-		configSlice = append(configSlice, config)
-	}
+	common.AC.WithLoadedConfigs(func(loadedConfigs map[string]integration.Config) {
```

Comment:
Since you do this same copy in 3 different places would it be worth also adding a helper function that returns a copy alongside `WithLoadedConfigs`?

Reply:
Good point.. I'll add a commit..